Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs

```diff
@@ -557,11 +558,14 @@ public override void Update(float frameTime)
 
                 active.CookTimeRemaining -= frameTime;
 
-                RollMalfunction((uid, active,microwave));
+                RollMalfunction((uid, active, microwave));
 
                 //check if there's still cook time left
                 if (active.CookTimeRemaining > 0)
                 {
+                    microwave.CurrentCookTimerElapsedTime = (float) Math.Floor(active.CookTimeRemaining);
+                    UpdateUserInterfaceState(uid, microwave);
```

Comment:
You should really just be using the active component for this as this is just going to spam the state out every tick which isn't ideal.

Ideally change CookTimeRemaining to the time when the cook finishes and just network that, that way client can predict it and you only need the 1 state rather than 30 x 5 states.

Reply:
No problem, but will be away for a week just fyi so update maybe slow.

Reply:
I have fully detached the countdown from the server and now it just passes the estimated time it will finish and the client counts down from then. Now uses timespan instead of a float.